IviDriverPython/1.0/Spec/IviDriverPython.md

```diff
@@ -0,0 +1,654 @@
+The compliance document for an IVI driver states whether the driver is available in a 64-bit version, a 32-bit version, or both.
+
+### Target Python Versions
+
+IVI-Python drivers shall target Python 3.8 or later.
+
+### IVI-Python Packages
+
+IVI-Python drivers shall be organized as a package, including a top-package `*__init__.py*` file.
+
+#### Terms used
+- `<Dist-pckg-name>` - Python Distribution Package Name. Example: `vendorxy-specan`
+- `<Top-pckg-name>` - driver's top-level package name. Example:  `vendorxy_specan`
+ 
+#### Distribution Package Naming
+The name of the distribution package for the driver shall follow the [Python naming guideline](https://packaging.python.org/en/latest/specifications/name-normalization/):
+The name should be lowercased with all runs of the characters ., -, or _ replaced with a single - character. This can be implemented in Python with the re module:
+
+`<Dist-pckg-name>` composition:
+- Variant 1: `<VendorPrefix>-<Instrument>` Example: `vendorxy-specan`
+- Variant 2: `<VendorPrefix>-<Instrument>-<CustomSuffix>`. Example: `vendorxy-mrtester-lte`
+
+Variant 2 can be used for cases where the instrument contains more sub-systems, and the driver is designed only for one of them.
+`<Instrument>` can represent a specific model, or a family of instruments.
+
+#### Top Package Naming
+
+`<Top-pckg-name>` should be derived from the distribution package name, and shall follow python package naming guideline - all lower case, words separated by underscore.
+For example, if a distribution package name is `vendorxy-specan` the top-level package should be called `vendorxy_specan`.
+
+#### Reference Property and Class Naming
+
+For reference exposed to the user, IVI-Python drivers shall follow the PEP-8 Python naming guidelines.
+
+### IVI-Python Driver Structure
+
+#### Terms used
+- `<Driver-identifier>` - driver's name and the top-level class name: Example: `VendorxySpecan`
+- `<Sub-pckg-name>` - further driver's packages: Example: `measurement_group`
+- `<Reference-property>` - further driver's interface access property: Example: `measurement`
+- `<Interface-class>` - further driver's interface class name. Example: `MeasurementCls`
+
+- [!NOTE]: need to discuss/specify the Sub-package-name, module names, class names to avoid ambivalence.
+
+The main class `<Driver-identifier>` defined in `root.py` shall include one or multiple properties `<Reference-property>`. A reference property is used to access the driver's structure. Each reference property returns an instance of `<Interface-class>`. `<Interface-class>` may in turn include further `<Reference-property>`(s), and so on. The hierarchy may be arbitrarily deep. Sub-modules within the driver may be named at the driver vendors discretion.
+
+Recommended instrument driver structure:
+
+[!NOTE] - structure for discussion on 'shall' and 'should', or whether even to suggest a driver structure and sub-package naming.
```

Comment:
It seems like the structure should probably by just a recommendation, some packages may want to use the `src/` layout.